assets/scripts/menus/LocaleDropdown.jsx

```diff
@@ -1,109 +1,68 @@
 import React from 'react'
 import PropTypes from 'prop-types'
-import { connect } from 'react-redux'
-import { injectIntl, intlShape } from 'react-intl'
-import { changeLocale } from '../store/actions/locale'
-import { trackEvent } from '../app/event_tracking'
+import { FormattedMessage } from 'react-intl'
 import LOCALES from '../../../app/data/locales.json'
 
-export class LocaleDropdown extends React.Component {
+const DEFAULT_LOCALE = 'en'
+
+export default class LocaleDropdown extends React.Component {
   static propTypes = {
-    intl: intlShape.isRequired,
     locale: PropTypes.string,
-    changeLocale: PropTypes.func,
-    /* eslint-disable react/no-unused-prop-types */
-    // These props _are_ used but linter can't tell
-    level1: PropTypes.bool.isRequired,
-    level2: PropTypes.bool.isRequired,
-    level3: PropTypes.bool.isRequired
-    /* eslint-enable react/no-unused-prop-types */
+    level: PropTypes.number,
+    selectLocale: PropTypes.func
   }
 
   static defaultProps = {
-    level1: false,
-    level2: false,
-    level3: false
-  }
-
-  constructor (props) {
-    super(props)
-
-    // Will be set by `getDerivedStateFromProps`
-    this.state = {
-      level: null
-    }
-  }
-
-  static getDerivedStateFromProps (nextProps) {
-    // The lowest level marked "true" takes priority.
-    let level = 4
-    if (nextProps.level3) level = 3
-    if (nextProps.level2) level = 2
-    if (nextProps.level1) level = 1
-
-    return { level }
-  }
-
-  componentDidMount () {
-    // Set the dropdown to the current language.
-    // If current language is not in the list, fallback to US English.
-    this.localeSelect.value = this.props.locale
-    if (!this.localeSelect.value) {
-      this.localeSelect.value = 'en'
-    }
-  }
-
-  onShow () {
-    trackEvent('Interaction', 'Open settings menu', null, null, false)
-  }
-
-  onChange = (event) => {
-    this.props.changeLocale(event.target.value)
+    locale: DEFAULT_LOCALE,
+    level: 4,
```

Comment:
Why do we default this to 4? Can we have have a comment here? Do other components pass in level1, level2 or level3 props that need to be changed as well?

Reply:
The different languages are classified as different levels of readiness, with "4" being "this is the default language" (English). Level 3 is any other "ready" language, level 2 is "close but testing" and level 1 is "not ready at all; just getting started".

I left a markdown file describing these here: https://raw.githubusercontent.com/streetmix/streetmix/master/app/data/locales.md

This is not obvious so I'll definitely leave a comment or figure out some other way of making this clearer.